nativelib/src/main/scala/scala/scalanative/runtime/Throwables.scala

```diff
@@ -75,31 +75,62 @@ object StackTrace {
     }
   }
 
+  @resolvedAtLinktime
+  private def hasDebugInfo: Boolean =
+    (LinktimeInfo.isMac || LinktimeInfo.isLinux) &&
+      !LinktimeInfo.is32BitPlatform &&
+      LinktimeInfo.sourceLevelDebuging.generateFunctionSourcePositions
+
   private def makeStackTraceElement(
       cursor: CVoidPtr,
       ip: CUnsignedLong
   ): StackTraceElement = {
-    val nameMax = 1024
-    val name = fromRawPtr[CChar](
-      calloc(
-        Intrinsics.castIntToRawSizeUnsigned(nameMax),
-        Intrinsics.sizeOf[CChar]
-      )
-    )
-    val offset = fromRawPtr[Long](Intrinsics.stackalloc[Long]())
-
-    unwind.get_proc_name(cursor, name, nameMax.toUSize, offset)
 
-    // Make sure the name is definitely 0-terminated.
-    // Unmangler is going to use strlen on this name and it's
-    // behavior is not defined for non-zero-terminated strings.
-    name(nameMax - 1) = 0.toByte
     val position =
-      if (LinktimeInfo.isMac && LinktimeInfo.sourceLevelDebuging.generateFunctionSourcePositions)
-        Backtrace.decodePosition(ip.toLong)
+      if (hasDebugInfo) Backtrace.decodePosition(ip.toLong)
       else Backtrace.Position.empty
-    try StackTraceElement(name, position)
-    finally free(name)
+
+    def withNameFromDWARF() = {
+      // linkageName has an extra "_" that we don't want in stack traces
+      def isScalaNativeMangledName =
+        position.linkageName(0) == '_'.toByte &&
+          position.linkageName(1) == '_'.toByte &&
+          position.linkageName(2) == 'S'.toByte
```

Comment:
We could consider usage of strncmp 
```suggestion
      def isScalaNativeMangledName = strncmp(position.linkageName, c"__SM", 4.toCSize) == 0
```
However, it would require us to add binding in `runtime.fii`

Reply:
Done ✅